pkg/webhook/vault_secret_manager_test.go

```diff
@@ -137,6 +157,18 @@ func TestVaultSecretManagerInjector_Inject(t *testing.T) {
 			want:    ExpectedKVv2,
 			wantErr: false,
 		},
+		{
+			name: "KVv3 Secret - with extra annotations",
+			args: args{
+				cfg: config.VaultSecretManagerConfig{Role: "flyte", KVVersion: config.KVVersion2, ExtraAnnotations: map[string]string{
```

Comment:
Can you add some more examples here to cover the various override scenarios?

Reply:
Added cases where users provide their annotation via pod template annotations and system override annotation.